txpipe/twopoint.py

```diff
@@ -55,20 +57,28 @@ def run(self):
         if self.comm:
             self.comm.Barrier()
 
-        # Get the number of bins from the
-        # tomography input file
-        nbins_source, nbins_lens  = self.read_nbins()
+        # the default is to run all bins, which is set in the config_options above
+        if self.config['source_bins'][0] == -1 and self.config['lens_bins'][0] == -1:
```

Comment:
We might also want to deal with the case where one of the sets of bins is empty, but I see you've put that as a TODO later on so we can deal with it later.

Reply:
sounds good, I can keep working on this development branch to include that feature